src/execution_plan/execution_plan.c

```diff
@@ -22,45 +22,108 @@
 #include <assert.h>
 #include <setjmp.h>
 
-static inline OpBase *_ExecutionPlan_LocateLeaf(OpBase *root) {
+/* Returns the left most leaf operation in the current segment. */
+static inline OpBase *_ExecutionPlan_LocateLeaf(OpBase *root, OpBase *prev_segment_head) {
+	if(root == prev_segment_head) return root->parent; // Don't recurse into the previous segment.
 	if(root->childCount == 0) return root;
-	return _ExecutionPlan_LocateLeaf(root->children[0]);
+	return _ExecutionPlan_LocateLeaf(root->children[0], prev_segment_head);
 }
```

Comment:
:+1: Good call, the `prev_segment_head` logic can be safely deleted! This had been a way to avoid accidentally recursing into a previous segment when we have multiple WITH clauses, but the projection-building logic has been improved such that this is no longer a concern.

Reply:
👍